### PR TITLE
Rename experimental JSON tests.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR renames the "experimental" JSON reader tests. These are now production grade and not experimental.

This task is tracked in https://github.com/rapidsai/cudf/issues/15537.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
@karthikeyann / @vuule: Could one of you help me understand this line?

https://github.com/rapidsai/cudf/blob/7b9e8158d38a250217b328ee005d9cf8581bec9f/cpp/tests/io/nested_json_test.cpp#L624

This boolean parameter labeled "Experimental" appears to control whether we call `device_parse_nested_json` or `host_parse_nested_json`. Do we still need both of those code paths (with tests)?

**Comment by vuule** [comment]
> @karthikeyann / @vuule: Could one of you help me understand this line?
> 
> https://github.com/rapidsai/cudf/blob/7b9e8158d38a250217b328ee005d9cf8581bec9f/cpp/tests/io/nested_json_test.cpp#L624
> 
> This boolean parameter labeled "Experimental" appears to control whether we call `device_parse_nested_json` or `host_parse_nested_json`. Do we still need both of those code paths (with tests)? Should we rename this flag from `Experimental` to `DeviceParse`?

IMO we can remove the host path, and thus this test. We now have a lot of tests of the new reader, I think we don't have to test this part of the implementation against the reference any more. If not, yes, renaming to DeviceParse sounds good to me
We do definitely need @karthikeyann's blessing to remove the test :)

**Comment by bdice** [comment]
Great. I renamed to `IsFullGPU` because that's how the variables are named in the tests. If @karthikeyann thinks we can remove the host parsing logic, I'll do that (and remove the tests) in a follow-up PR.

**Review by davidwendt — APPROVED** [review]

**Comment by karthikeyann** [comment]
Yes. We could remove the host parsing logic. It may not be in sync with device code with a lot of recent changes in JSON tokenizer, and tree algorithms. @elstehle shall we remove host parsing code and their tests?

**Comment by davidwendt** [comment]
Did you merge this with only **one** cpp review approval?

**Comment by bdice** [comment]
> Did you merge this with only **one** cpp review approval?

Oops. @davidwendt I thought you were the second approver. My apologies.

**Comment by bdice** [comment]
@vuule and/or @karthikeyann feel free to take a look over this. I thought Vukasin had approved first and David was second, but I now see that I imagined the first ✅ where there was none. I hope we don’t need to revert but I’ll be happy to apply any follow-ups in another PR if needed.

**Comment by vuule** [comment]
How could you do this @bdice ?!
Well, no harm done, I would approve as-is. Just nobody tell Greg.

**Pull request merged**
